### ISSUE TITLE
Having a folder named 'switch' at the root of your SD card prevents ftpd from working

### ISSUE THREAD

**Issue by briankendall** [mtheall/ftpd#80]
If you have a folder named `switch` or `SWITCH` at the root of your SD card then any time you try to list the root directory you get a `550 unavailable` error. On the switch ftpd will print an error message that looks something like: `stat '/switch': no such file or directory`. This prevents most FTP clients from being able to connect or do much of anything.

When I removed that folder it cleared up the issue. ftpd created a 0 byte file named `SWITCH` at the root of my SD card, which is presumably why the directory was interfering. (I'm not sure what this file is or why it's important.)

**Comment by mtheall** [comment]
When this situation occurs, what happens if your client sends a `NLST /` command? How about `STAT /switch`?

**Comment by briankendall** [comment]
`NLST /` successfully displays the contents of the root of my SD card, including the `SWITCH` folder, though on my SD card it's named with all lower case, just in case that detail matters.

`STAT /switch` returns `550 No such file or directory`.

I'm running firmware 6.0.1 and SX OS 2.0.1.

**Comment by briankendall** [comment]
I just managed to fix everything. Whatever was going wrong, I don't think it was the fault of ftpd -- something must have been corrupt about my SD card. I had previously set up my SD card on macOS and copied files over that way, and even though it was formatted fat32 I think (but am not 100% sure)  macOS did something that wasn't compatible with the Switch. I just deleted the `/switch` folder and a bunch of other extraneous files in Windows 10, at which point ftpd allowed me to browse my files correctly and didn't see an extraneous file named `SWITCH` at the root of my SD card. I then used the Homebrew Store to download some apps, it created a new `/switch` folder, and everything is working swimmingly. A few other homebrew apps that crashed the switch before are now working properly.

I don't know what's different about this new `/switch` folder -- as near as I can tell it has all of the same attributes set. But as long as things are working, I'm fairly happy. I just wish I understood what went wrong in the first place!
